sklearn/metrics/tests/test_ranking.py

```diff
@@ -891,35 +891,41 @@ def test_binary_clf_curve_zero_sample_weight(curve_func):
         assert_allclose(arr_1, arr_2)
 
 
-def test_precision_recall_curve():
+@pytest.mark.parametrize("drop", [True, False])
+def test_precision_recall_curve(drop):
     y_true, _, y_score = make_prediction(binary=True)
-    _test_precision_recall_curve(y_true, y_score)
+    _test_precision_recall_curve(y_true, y_score, drop)
 
     # Make sure the first point of the Precision-Recall on the right is:
     # (p=1.0, r=class balance) on a non-balanced dataset [1:]
-    p, r, t = precision_recall_curve(y_true[1:], y_score[1:])
+    p, r, t = precision_recall_curve(y_true[1:], y_score[1:], drop_intermediate=drop)
     assert r[0] == 1.0
     assert p[0] == y_true[1:].mean()
 
     # Use {-1, 1} for labels; make sure original labels aren't modified
     y_true[np.where(y_true == 0)] = -1
     y_true_copy = y_true.copy()
-    _test_precision_recall_curve(y_true, y_score)
+    _test_precision_recall_curve(y_true, y_score, drop)
     assert_array_equal(y_true_copy, y_true)
 
     labels = [1, 0, 0, 1]
     predict_probas = [1, 2, 3, 4]
-    p, r, t = precision_recall_curve(labels, predict_probas)
-    assert_array_almost_equal(p, np.array([0.5, 0.33333333, 0.5, 1.0, 1.0]))
-    assert_array_almost_equal(r, np.array([1.0, 0.5, 0.5, 0.5, 0.0]))
-    assert_array_almost_equal(t, np.array([1, 2, 3, 4]))
+    p, r, t = precision_recall_curve(labels, predict_probas, drop_intermediate=drop)
+    if drop:
+        assert_array_almost_equal(p, np.array([0.5, 0.33333333, 1.0, 1.0]))
```

Comment:
and we can make the writing lighter by removing the `np.array`. The conversion will be implicitly done in the assertion.